hms-app/src/components/HackathonSelectDropdown.tsx

```diff
@@ -27,38 +38,55 @@ export default function HackathonSelectDropdown(setHackathonID: Props) {
     )
   }
 
-  const hackathonMap = hackathonList.map((hackathon, index) => ({
-    value: hackathon.id,
-    label:
-      hackathon.title +
-      ' ' +
-      new Date(hackathon.startDate).toLocaleDateString() +
-      '-' +
-      new Date(hackathon.endDate).toLocaleDateString(),
-  }))
+  function getHackathonsSelectItems(): SelectItem[] {
+    switch (context) {
+      case HackathonDropdownMode.Archive:
+        return hackathonList
+          .filter((hackathon) => new Date(hackathon.endDate) < new Date(today))
```

Comment:
now this should for sure work without the `new Date()` (around hackathon.endDate and for sure around today)